tests/system/projects/test_project.py

```diff
@@ -1244,6 +1244,42 @@ def test_project_default_function_node_selector_using_igz_mgmt(self):
         }
         self._create_and_validate_project_function_with_node_selector(project)
 
+    def _create_and_validate_spark_function_with_project_node_selectors(self, project):
+        function_name = "spark-function"
+        function_label_name, function_label_val = "kubernetes.io/os", "linux"
+        function_override_label, function_override_val = "kubernetes.io/hostname", ""
+        file_name = "spark.py"
+
+        self._files_to_upload.append(file_name)
+        self._upload_code_to_cluster()
+        code_path = os.path.join(self.remote_code_dir, file_name)
+        spark_function = mlrun.new_function(
+            name=function_name,
+            kind="spark",
+            command=code_path.replace("v3io:///", "/v3io/"),
+        )
+        spark_function.with_igz_spark()
+        spark_function.with_driver_limits(cpu="1300m")
+        spark_function.with_driver_requests(cpu=1, mem="512m")
+
+        spark_function.with_executor_limits(cpu="1400m")
+        spark_function.with_executor_requests(cpu=1, mem="512m")
+
+        node_selector = {
+            function_label_name: function_label_val,
+            function_override_label: function_override_val,
+        }
+
+        spark_function.with_node_selection(node_selector=node_selector)
```

Comment:
use executer and driver since this is deprecated

Reply:
I guess you are right, but I also wanted to test that the changes to `with_node_selection` works as expected now, do you think it's okay not to test it and just  use the driver and executor ones?